state/syncer/baseAccountsSyncer.go

```diff
@@ -214,6 +214,30 @@ func (b *baseAccountsSyncer) GetSyncedTries() map[string]common.Trie {
 	return clonedMap
 }
 
+func markStorerAsSyncedAndActive(mainTrie common.Trie) {
+	trieStorageManager := mainTrie.GetStorageManager()
+
+	epoch, err := trieStorageManager.GetLatestStorageEpoch()
+	if err != nil {
+		log.Error("getLatestStorageEpoch error", "error", err)
+	}
+
+	err = trieStorageManager.Put([]byte(common.TrieSyncedKey), []byte(common.TrieSyncedVal))
+	if err != nil {
+		log.Error("error while putting trieSynced value into main storer after sync", "error", err)
+	}
+
+	lastEpoch := epoch - 1
```

Comment:
if `trieStorageManager.GetLatestStorageEpoch()` returns error, the `epoch` will be set to `0`. therefore, `lastEpoch` will be 0 - 1 so max uint 32, and the check will pass

Reply:
If `epoch == 0` , `lastEpoch` will be overwritten with 0, so it will work as expected.